src/benchmarklib/benchmark_runner.cpp

```diff
@@ -90,8 +82,15 @@ void BenchmarkRunner::run() {
       pipeline.get_result_table();
     }
   }
+}
+
+BenchmarkRunner::~BenchmarkRunner() {
+  if (CurrentScheduler::is_set()) {
```

Comment:
BenchmarkRunner::run() isn't guaranteed to wait until all queries finished?

Reply:
I think this is more of a general precaution.

Reply:
A general precaution that would potentially hide a possible bug in BenchmarkRunner::run(), right?

Reply:
Addressed in #1597

Reply:
just realised that this was there before. Alright, I rest my case